src/app/modules/navigator/navigator.component.ts

```diff
@@ -1,15 +1,76 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ChangeDetectorRef, Output, OnDestroy } from '@angular/core';
+import { TreeNode } from '../tree-viewer/tree-node';
+import { TreeViewerConfig } from '../tree-viewer/tree-viewer-config';
+import { TestCaseService, CallTreeNode } from '../test-case-service/default.test.case.service';
+import { MessagingService } from '@testeditor/messaging-service';
+import { Subscription } from 'rxjs/Subscription';
+
+const NAVIGATION_OPEN = 'navigation.open';
+interface NavigationOpenPayload {
+  name: string;
+  path: string;
+}
+const EMPTY_TREE: TreeNode = { name: '<empty>', children: [] };
 
 @Component({
   selector: 'app-navigator',
   templateUrl: './navigator.component.html',
   styleUrls: ['./navigator.component.css']
 })
-export class NavigatorComponent implements OnInit {
+export class NavigatorComponent implements OnInit, OnDestroy {
```

Comment:
we should probably consider renaming this (maybe TestNavigatorComponent?), esp. the selector is very generic (doesn't need to be renamed now, if it causes too much hassle).
I'm assuming here that this component / module will contain domain-specific code, while only the TreeView will be truly generic, right?